internal/keepers/util.go

```diff
@@ -138,6 +142,29 @@ func shuffledDedupedKeyList(attributed []types.AttributedObservation, key [16]by
 		return nil, fmt.Errorf("%w: observation dedupe", err)
 	}
 
+	// TODO: another hacky solution assuming upkeep key structure
```

Comment:
same comment. let's collect all places where we do the split of key into single function
and document the hack in a ticket and link it here